include/javabind/binding.hpp

```diff
@@ -374,4 +375,46 @@ namespace javabind
             return *this;
         }
     };
+
+    struct EnumBinding {
+        std::vector<std::string_view> names;
+
+        void add(std::string_view name)
+        {
+            if (!contains(name)) {
+                names.push_back(name);
+            }
+        }
+
+        bool contains(std::string_view name) const
+        {
+            return std::find(names.begin(), names.end(), name) != names.end();
+        }
+    };
+
+    struct EnumBindings {
+        inline static std::map< std::string_view, EnumBinding > value;
+    };
+
+    /**
+     * Represents an enum class in Java.
+     */
+    template <typename T>
+    struct enum_class
+    {
+        enum_class()
+        {
+            EnumBindings::value.emplace(arg_type_t<T>::class_name, EnumBinding {});
+        }
+
+        enum_class<T>& value(T native_value, std::string_view java_name)
+        {
+            EnumBindings::value.at(arg_type_t<T>::class_name).add(java_name);
```

Comment:
Do we need `at()` here (instead of `operator[]`)? The constructor for `enum_class` guarantees the value collection exists.

Reply:
My reasoning is that `at()` is preferred over `operator[]` because the value should already have been created by the constructor. What do you prefer, keep `at()` or change to `operator[]`?

Reply:
Yeah, I totally forgot about this seemingly strange behavior of `std::map<K,V>` on `operator[]` that it inserts a new entry if the key doesn't exist. `at()` is definitely the better choice.